core/ordersync/ordersync.go

```diff
@@ -477,7 +490,7 @@ type FirstRequestsForSubprotocols struct {
 func (s *Service) createFirstRequestForAllSubprotocols() (*rawRequest, error) {
 	metadata := []json.RawMessage{}
 	for _, subprotocolString := range s.SupportedSubprotocols() {
```

Comment:
We should just iterate through `subprotocolSet` directly instead of iterating through `SupportedSubprotocols` and looking up the corresponding subprotocol in the set.

```suggestion
	for subprotocolId, subprotocol := range s.subprotocolSet {
```

Reply:
I don't think we can do this. The metadata needs to correspond to the prioritization of the subprotocols. Wouldn't this be a random order?

Reply:
👍 You're right. I was commenting on the old version which iterated through `SupportedSubprotocols` which is also in random order. The new version is correct.